### ISSUE TITLE
why is pgtype.JSONB PreferredResultFormat == TextFormatCode instead of BinaryFormatCode? 

### ISSUE THREAD

**Issue by Ashtonian** [jackc/pgtype#168]
I just noticed the JSONB types PreferredResultFormat is set to TextFormatter which avoids binary decoding, is this intentional?

**Comment by jackc** [comment]
Yes. The only difference between text and binary format is the binary format is prefixed by binary `1`. Using the text format makes it the result more flexible. Also, returning text values in the binary format can be significantly slower. See https://www.postgresql.org/message-id/CAMovtNoHFod2jMAKQjjxv209PCTJx5Kc66anwWvX0mEiaXwgmA%40mail.gmail.com
